python/paddle/v2/framework/tests/test_activation_op.py

```diff
@@ -144,6 +144,24 @@ def test_check_grad(self):
         self.check_grad(['X'], 'Y', max_relative_error=0.02)
 
 
+class TestELU(OpTest):
+    def setUp(self):
+        self.op_type = "elu"
+        x = np.random.uniform(-3, 3, [4, 4]).astype("float32")
+        alpha = 1.
+        self.inputs = {'X': x}
+        self.attrs = {'alpha': alpha}
+        self.outputs = {
+            'Y': np.maximum(0, x) + np.minimum(0, alpha * (np.exp(x) - 1))
```

Comment:
Thanks for the reference, I'm a little bit confused by the line `x[np.abs(x) < 0.005] = 0.02` before and #4120 explains the motivation.

However, ELU has a quite "smooth" negative part, so this modification may be unnecessary here. In fact, `elu(x=-0.005)` gives -0.0049875208073176802, which makes the relative numeric gradient error less than 2e-5 and is small enough to pass normal gradient check.
Here is the code I used to experiment.
```
import matplotlib.pyplot as plt
import numpy as np
def elu(x):
    return np.maximum(0, x) + np.minimum(0, 1. * (np.exp(x) - 1))
x = np.linspace(-0.5, 0.5, num=100)
y = elu(x)
plt.plot(x,y)
```
![figure_1](https://user-images.githubusercontent.com/22702369/30959527-0e18f948-a473-11e7-842b-ba6d90cf2df7.png)

Reply:
But on the other hand, I can make another PR if you think we should make this modification `x[np.abs(x) < 0.005] = 0.02` a common practice in unit test.

Reply:
Yes, elu is actually very smooth. 0 is a non-differentiable point in elu. We'd better to filter it in out test data.
This line
```
x[np.abs(x) < 0.005] = 0.02
```
will filter these potential points and set them to 0.02.(0.02-0.005 = 0.015;0.02+0.005=0.025, both of 0.015 and 0.025 are larger than 0)

Reply:
As long as we are talking about the standard `elu`, i.e. alpha=1, then it is differentiable at point 0 since both left derivate and right derivate at 0 equal to 1. The "piecewise" form of the function only makes it non-differentiable on the derivative function at 0, i.e. the second order derivative.

Reply:
@QiJune I didn't add the filtering code to the new commit, since as we discussed it is not necessary. However, I add a comment to explain the reason.